### PR TITLE
feat: Add `boundary_group` data source

### PR DESCRIPTION
**Pull request by moduli**
This PR adds a `boundary_group` data source
```
data "boundary_group" "global_group" {
  name = "admin"
}

data "boundary_group" "org_group" {
  name     = "username"
  scope_id = data.boundary_scope.org.id
}
```

<details>
<summary>Test</summary>

```
resource "boundary_user" "user1" {
  name     = "user1"
  scope_id = "global"
}

resource "boundary_user" "user2" {
  name     = "user2"
  scope_id = "global"
}

resource "boundary_group" "group" {
  name       = "group"
  scope_id   = "global"
  member_ids = [boundary_user.user1.id, boundary_user.user2.id]
}

data "boundary_group" "group" {
  depends_on = [boundary_group.group]
  name       = "group"
}

output "group" {
  value = data.boundary_group.group
}


Outputs:

group = {
  "description" = ""
  "id" = "g_WlBSAYpsn3"
  "member_ids" = toset([
    "u_oPPhtdmDSu",
    "u_qDLvPFPCWS",
  ])
  "name" = "group"
  "scope" = tolist([
    {
      "description" = "Global Scope"
      "id" = "global"
      "name" = "global"
      "parent_scope_id" = ""
      "type" = "global"
    },
  ])
  "scope_id" = "global"
}
```
</details>

### CODE REVIEW

**Comment by grantorchard** [comment]
Excited for this one! Just had a need for a group datasource and decided to quickly check if someone had started work on it already 👍

File: examples/data-sources/boundary_group/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Retrieve a user from the global scope
+
```
Comment (louisruch) [inline examples/data-sources/boundary_group/data-source.tf:2]: nit: remote white space

File: examples/data-sources/boundary_group/data-source.tf [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+data "boundary_group" "global_group" {
+  name = "admin"
+}
+
+# User from an org scope
+
```
Comment (louisruch) [inline examples/data-sources/boundary_group/data-source.tf:8]: nit: remove white space

File: internal/provider/const.go [hunk 3]
```diff
@@ -51,4 +51,6 @@ const (
 	PrimaryAccountIdKey = "primary_account_id"
 	ScopeKey            = "scope"
 	ParentScopeIdKey    = "parent_scope_id"
+
+	GroupMemberIdsKey = "member_ids"
```
Comment (louisruch) [inline internal/provider/const.go:55]: We generally like to add comments to exported variables so godocs has a comment for it. I see we did not do that for some of the above consts, but would be good to add it here

**Pull request merged**